provider/lxd/environ_broker.go

```diff
@@ -324,6 +324,12 @@ func (env *environ) AllInstances(ctx context.ProviderCallContext) ([]instances.I
 	return instances, errors.Trace(err)
 }
 
+// AllRunningInstances implements environs.InstanceBroker.
+func (env *environ) AllRunningInstances(ctx context.ProviderCallContext) ([]instances.Instance, error) {
+	// We can only get Alive containers from lxd api which means that "all" is the same as "running".
```

Comment:
This comment isn't quite right - the AliveContainers method in container/lxd.Server includes stopping and stopped containers - there's nothing stopping us from adding another one which doesn't.

https://github.com/juju/juju/blob/develop/container/lxd/container.go#L148